### PR TITLE
readline: Don't echo input except VT100 command

### PR DESCRIPTION
**Pull request by no1wudi**
## Summary
Let's echo character by serial driver, this will imporve the experience of telnetd, which will display input command and newline twice.
And also enhance the compatiblity with some third party linux readline based application (like WAMR repl).
## Impact
nsh
## Testing
nsh over serial and telnetd

### CODE REVIEW

File: system/readline/readline_common.c [hunk 1]
```diff
@@ -708,11 +708,6 @@ ssize_t readline_common(FAR struct rl_common_s *vtbl, FAR char *buf,
           buf[nch++] = '\n';
           buf[nch]   = '\0';
 
-#ifdef CONFIG_READLINE_ECHO
```
Comment (xiaoxiang781216) [inline system/readline/readline_common.c:711]: should we remove CONFIG_READLINE_ECHO?

Comment (no1wudi) [reply system/readline/readline_common.c:711]: Maybe should rename it to represent its function now (cmd history/backspace etc)

Comment (xiaoxiang781216) [reply system/readline/readline_common.c:711]: understand, let's continue use CONFIG_READLINE_ECHO represent other echo.

**Review by xiaoxiang781216 — APPROVED** [review]

**Comment by masayuki2009** [comment]
@no1wudi 
I noticed that nsh does not show input characters when tested with qemu (e.g. rv-virt:smp64)
If I revert this PR, it works again.

**Comment by no1wudi** [comment]
> 

@masayuki2009 Sorry, This PR should works whti https://github.com/apache/nuttx/pull/8691, but it merged late.

**Comment by masayuki2009** [comment]
>@masayuki2009 Sorry, This PR should works whti https://github.com/apache/nuttx/pull/8691, but it merged late.

Thanks for the information. 
I confirmed that the issue has been resolved with the above PR.

**Pull request merged**